### ISSUE TITLE
initialize PhoneNumberPrefixWidget with initial don't make effect

### ISSUE THREAD

**Issue by boy2000-007man** [stefanfoulis/django-phonenumber-field#202]
Hi,
Thanks for the work.:)

When setting the default region number with parameter initial, I find something strange.
My code is shown below:
```
phone = PhoneNumberField(
    label=_('phone number'),
    required=True,
    widget=PhoneNumberPrefixWidget(
        initial='CN',
        attrs={'placeholder': _('Phone number')}))
```
The default prefix is still empty.:(
I trace the code and find PhonePrefixSelect.render() won't be accessed during runtime, which should be to pass the default prefix.
Later I find the default value of the prefix may be fixed by changing the PhoneNumberPrefixWidget.decompress()'s return value.
I don't understand what happens, does latest Django v2.0.2 change the rendering logic so that the PhonePrefixSelect.render() loses its function?
Or there is something wrong with my code I don't notice?

Thank You

**Comment by quevon24** [comment]
Worked form me this way:

cellphone = PhoneNumberField(
        widget=PhoneNumberPrefixWidget(attrs={'placeholder': _(u'Cellphone'), 'class': "form-control"}),
        label=_(u'Cellphone number'),
        required=False,
        initial='+52'
    )

**Comment by boy2000-007man** [comment]
Thanks a lot. I'll try it later. Your solution seems using the CharFields's initial keyword. And I think the PhoneNumberPrefixWidget's own initial keyword may not correctly be implemented.

**Comment by QFXC** [comment]
This is still an issue. I am using Django version 2.0.9. I had to come up with this hacky way of getting a default/initial value.

`
# forms.py
from .widgets import CustomPhoneNumberPrefixWidget
class AccountProfilePhoneForm(AccountModelForm):
    class Meta:
        model = AccountProfile
        fields = AccountProfile.PHONE_FORM_MINIMUM
        widgets = {'phone_num': CustomPhoneNumberPrefixWidget(initial=('+1', 'United States +1'))}

# widgets.py
from phonenumber_field.widgets import (
    PhonePrefixSelect, Select, translation, settings, 
    _COUNTRY_CODE_TO_REGION_CODE, Locale, TextInput, PhoneNumber, MultiWidget,
)

class CustomPhonePrefixSelect(PhonePrefixSelect):
    initial = None

    def __init__(self, initial=None):
        choices = [("", "---------")]
        language = translation.get_language() or settings.LANGUAGE_CODE
        if language:
            locale = Locale(translation.to_locale(language))
            for prefix, values in _COUNTRY_CODE_TO_REGION_CODE.items():
                prefix = "+%d" % prefix
                # if initial and initial in values:
                #     self.initial = prefix
                for country_code in values:
                    country_name = locale.territories.get(country_code)
                    if country_name:
                        choices.append((prefix, "%s %s" % (country_name, prefix)))
        sorted_choices = sorted(choices, key=lambda item: item[1])
        if initial:
            if not isinstance(initial, tuple):
                raise Exception(
                    "Custom Error: The initial argument must be a 2-tuple. "
                    r"For example, ('+1', 'United States +1')"
                )
            sorted_choices[0] = initial

        Select.__init__(self=self, choices=sorted_choices)



class CustomPhoneNumberPrefixWidget(MultiWidget):
    """
    A Widget that splits phone number input into:
    - a country select box for phone prefix
    - an input for local phone number
    """

    def __init__(self, attrs=None, initial=None):
        widgets = (CustomPhonePrefixSelect(initial), TextInput())
        super(CustomPhoneNumberPrefixWidget, self).__init__(widgets, attrs)

    def decompress(self, value):
        if value:
            if type(value) == PhoneNumber:
                if value.country_code and value.national_number:
                    return ["+%d" % value.country_code, value.national_number]
            else:
                return value.split(".")
        return [None, ""]

    def value_from_datadict(self, data, files, name):
        values = super(CustomPhoneNumberPrefixWidget, self).value_from_datadict(
            data, files, name
        )
        if all(values):
            return "%s.%s" % tuple(values)
        return ""
`

**Comment by jdufresne** [comment]
Fixed in 29b4866c06cfce3edafd813365d8f32c28efe499.
